### PR TITLE
Revert from a ReplicatedProducer to a legacy EDProducer

### PR DESCRIPTION
**Pull request by kutschke**
This is a, hopefully, temporary solution to Offline issues #744 and #745  that were uncovered when working on issue #482.

Using Production/JobConfig/validation/stoppedMuonsSingleStageMT.fcl  I ran 4 test jobs, 2 with the original code and with this PR.  All were 250000 events.  The two with this PR produced identical results while the two originals differed as described in issue #482 .

This PR could slow down the real time performance of the two jobs because of less parallelism.  So I did a measurement and the change is not large enough to be important.    The execution times for the 4 jobs were

Before this PR:
TimeReport CPU = 9158.873317 Real = 1878.142641
TimeReport CPU = 9288.815459 Real = 1904.679477

 Mean and standard deviation:  1891 +/- 23

After this PR:
TimeReport CPU = 9301.399397 Real = 1906.883257
TimeReport CPU = 9358.623558 Real = 1916.216917

 Mean and standard deviation:   1911 +/- 7

The difference between the two means is about 1%, with a significance of about 1 sigma.  I believe this is acceptable.

### CODE REVIEW

**Review by resnegfk — APPROVED** [review]
In my opinion, the very low additional CPU cost is fully acceptable to regain the reproducibility.

File: EventGenerator/src/PrimaryProtonGun_module.cc [hunk 1]
```diff
@@ -1,6 +1,8 @@
 
 /*
-  This is a Replicated Module.
+  This was a Replicated Module and was reverted to Legacy on Mar 27, 2002
```
Comment (goodenou) [inline EventGenerator/src/PrimaryProtonGun_module.cc:3]: Typo here. You probably want this to say Mar 27, 2022.

**Review by goodenou — APPROVED** [review]
I recommend a fix to the date in line 3, but otherwise this looks good.

File: EventGenerator/src/PrimaryProtonGun_module.cc (hunk 1 above)
Comment (kutschke) [reply EventGenerator/src/PrimaryProtonGun_module.cc]: Thanks.  It's fixed.

**Review by brownd1978 — APPROVED** [review]

**Pull request merged**